### PR TITLE
Wr/sandbox create

### PR DESCRIPTION
**Pull request by WillieRuemmele**
@W-9805922@

adds `createSandbox` method to `Org` to create a sandbox from a production Org with available licenses
adds UTs for `createSandbox` and helper methods
adds `OrgTypes` and other enum types

### CODE REVIEW

**Review by mshanemc — APPROVED** [review]
none of the comments are blocking.

My only thought after reading this was...does any of this have to be on the already large Org class?

I'm imagining a `Sandbox` class might be cleaner (you could pass in an org to its only exported method?).  A lot of `Org` get constructed that are never going to manipulate sandboxes.

File: src/org.ts [hunk 1]
```diff
@@ -34,10 +34,58 @@ import { OrgUsersConfig } from './config/orgUsersConfig';
...
 import { Connection, SingleRecordQueryErrors } from './connection';
 import { Global } from './global';
+import { Lifecycle } from './lifecycleEvents';
 import { Logger } from './logger';
 import { SfdxError } from './sfdxError';
 import { fs } from './util/fs';
 import { sfdc } from './util/sfdc';
+import { WebOAuthServer } from './webOAuthServer';
+
+export enum OrgTypes {
```
[the remark this answers is not in the document]
Comment (WillieRuemmele) [reply src/org.ts:44]: it's used in the plugin, but I thought it made more sense to have definitive types here

File: src/org.ts [hunk 2]
```diff
@@ -735,6 +815,179 @@ export class Org extends AsyncCreatable<Org.Options> {
...
+    let pollFinished = false;
+    let waitingOnAuth = false;
+    const sandboxInfo = await this.sandboxSignupComplete(sandboxProcessObj);
+
+    if (sandboxInfo) {
+      try {
+        this.logger.debug('sandbox signup complete with %s', sandboxInfo);
+        await this.writeSandboxAuthFile(sandboxProcessObj, sandboxInfo);
+        pollFinished = true;
+      } catch (err) {
```
[the remark this answers is not in the document]
Comment (WillieRuemmele) [reply src/org.ts:894]: hmmm, I wasn't seeing any squiggles... I added the `?` operator to be safe

File: src/org.ts [hunk 3]
```diff
@@ -900,19 +916,26 @@ export class Org extends AsyncCreatable<Org.Options> {
...
         // emit the signup progress of the sandbox and query the production org again after waiting the interval
-        await lifecycle.emit(SandboxEvents.EVENT_STATUS, {
-          sandboxProcessObj,
-          interval: pollInterval.seconds,
-          retries,
-          waitingOnAuth,
-        });
-        await sleep(pollInterval);
+        await Promise.all([
+          await lifecycle.emit(SandboxEvents.EVENT_STATUS, {
```
Comment (mshanemc) [inline src/org.ts:920]: you don't have to await them inside `Promise.all`

**Review by mshanemc — APPROVED** [review]

**Pull request merged**